workers/unity/Packages/com.improbable.gdk.core/Worker/DefaultWorkerConnector.cs

```diff
@@ -10,62 +12,151 @@ public abstract class DefaultWorkerConnector : WorkerConnector
         /// </summary>
         public bool UseExternalIp;
 
-        /// <summary>
-        ///     Determines whether to connect via the locator.
-        /// </summary>
-        /// <returns>True, if should connect via the Locator, false otherwise.</returns>
-        protected override bool ShouldUseLocator()
+        protected override ConnectionService GetConnectionService()
```

Comment:
Since this is getting more complicated now, why not accept a command line argument for `connectionService`? Then we assert the parameters using that. The logic within here is too magical at this point.

(Can be done in another PR, shall I make a ticket for it?)

Reply:
The launcher does not pass that argument in... so we would need to check for both cases:
1. it is available
2. it is not available

Reply:
Oh I see... will the launcher always be using a particular kind of service?

Reply:
it provides everything to use the alpha locator flow, but you can also use the normal locator flow

Reply:
discussed offline, looks good